Lib/idlelib/idle_test/run_coverage.py

```diff
@@ -0,0 +1,169 @@
+
+    if not args.no_html:
+        print('Coverage report available at htmlcov/index.html')
+
+
+def listfiles(path):
+    return [ent.name for ent in os.scandir(path) if ent.is_file()]
+
+
+def listdirs(path):
+    return [ent.name for ent in os.scandir(path) if ent.is_dir()]
+
+
+def find_idlelib_path():
+    curpath = pathlib.Path(os.getcwd())
+    if curpath.name == 'idlelib':
+        idlelib_path = curpath
+    elif curpath.parent.name == 'idlelib':
+        idlelib_path = curpath.parent
+    elif 'idlelib' in listdirs(curpath):
+        idlelib_path = curpath.joinpath('idlelib')
+    elif (
+        'Lib' in listdirs(curpath) and
+        curpath.joinpath('Lib', 'idlelib').is_dir()
+    ):
+        idlelib_path = curpath.joinpath('Lib', 'idlelib')
+    else:
+        raise Exception('Failed to find idlelib path.\n'
+                        'Run in the base source dir or in Lib/idlelib.')
+    return idlelib_path
+
+
+def get_module_names(idlelib_path):
+    test_module_names = {
+        filename[len('test_'):-len('.py')]
+        for filename in listfiles(idlelib_path.joinpath('idle_test'))
+        if filename.startswith('test_') and filename.endswith('.py')
+    }
+    module_names = {
+        filename[:-len('.py')]
+        for filename in listfiles(idlelib_path)
+        if filename.endswith('.py')
+    }
+    return set(test_module_names) & set(module_names)
+
+
+def ensure_venv(sources_root_path):
+    venv_path = sources_root_path.joinpath(
+        'Lib', 'idlelib', 'idle_test', 'coverage_venv')
+    venv_python_path = (
+        venv_path.joinpath('Scripts', 'python.exe')
+        if sys.platform == 'win32' else
+        venv_path.joinpath('bin', 'python')
+    )
+    built_python_path = sources_root_path.joinpath(
+        'python.bat' if sys.platform == 'win32' else
+        'python.exe' if sys.platform == 'darwin' else
+        'python'
+    )
+    if not venv_path.is_dir():
+        subprocess.run([built_python_path, '-m', 'venv', venv_path])
+        subprocess.run([venv_python_path, '-m', 'pip', 'install', 'coverage'])
+    return venv_path, venv_python_path
+
+
+@contextlib.contextmanager
+def coveragerc_replacement():
+    coveragerc_path = pathlib.Path('.coveragerc')
+    orig_coverage_rc = None
+    if coveragerc_path.is_file():
+        with coveragerc_path.open('r', encoding='utf-8') as f:
+            orig_coverage_rc = f.read()
+    try:
+        with coveragerc_path.open('w', encoding='utf-8') as f:
+            f.write(textwrap.dedent('''\
+                [run]
+                branch = True
+                cover_pylib = True
+
+                [report]
+                # Regexes for lines to exclude from consideration
+                exclude_lines =
+                    # Don't complain if non-runnable code isn't run:
+                    if 0:
+                    if False:
+                    if __name__ == .__main__.:
+
+                    .*# htest #
```

Comment:
The readme is behind the file I actually use.
```suggestion

                    .*# pragma: no cover
                    .*# pragma: no branch

                    .*# htest #
                    if not (_htest or _utest):
```
The pragmas are generic to coverage; 'no cover' is currently used in idlelib.

The suggested coveragerc should be a separate file in idle_test/.  If I knew how to put a symlink in /dev, where I use it, I would to avoid the 2 copies issues.